apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/mapper/OnboardingMapper.java

```diff
@@ -0,0 +1,39 @@
+package it.pagopa.selfcare.onboarding.mapper;
+
+import it.pagopa.selfcare.onboarding.dto.OnboardingAggregateOrchestratorInput;
+import it.pagopa.selfcare.onboarding.entity.Aggregator;
+import it.pagopa.selfcare.onboarding.entity.Institution;
+import it.pagopa.selfcare.onboarding.entity.Onboarding;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+import java.util.Objects;
+
+@Mapper(componentModel = "cdi")
+public interface OnboardingMapper {
+
+    @Mapping(target = "institution", expression = "java(mapInstitution(input.getAggregate(), input.getInstitution()))")
+    @Mapping(target = "aggregator", expression = "java(mapAggregator(input.getInstitution()))")
```

Comment:
you could avoid to create a method for mapping aggregator....you could use `@Mapping(target = "aggregator", source = "institution")`

Reply:
We need to set institutionType and Origin from aggregator. I'll try to replace with @Mapping